iml-gui/crate/src/page/mod.rs

```diff
@@ -96,7 +97,13 @@ impl Page {
             Self::Target(m) => format!("Target: {}", &m.target.name),
             Self::Users => "Users".into(),
             Self::User(m) => format!("User: {}", &m.user.username),
-            Self::Volumes(_) => "Volumes".into(),
+            Self::Volumes(m) => {
```

Comment:
It's probably worth making a new page variant rather than switching over `Page::Volumes` and checking the internal model. 

I.E. creating `Page::ServerVolumes`, which parallels `Route::ServerVolumes`.

Reply:
I was thinking about `Self::User(m) => m.title()` (in the comment above). This would let make all fields private.

Reply:
Independent of making something like a `Title` trait that page models implement, I still think we should not reuse pages for different routes.

Reply:
I am 50/50 on this. I agree that this is a kind of different page, but on the other side it is the same page with the same model and the code. Technically, we reuse other pages (like Server) for different URLs (server id). And there is another extreme example - the NotFound page :)

Reply:
Where does server page get reused?

Reply:
I mean we use it with different models with different server id :) Just like other pages: user, volume, filesystem.

Reply:
Sort of; What I mean is it's largely been a 1:1 mapping between the `Route` enum and a `Page` enum. 

We are reusing the `Page::Volume` variant for both the `Route::Volumes` and `Route::ServerVolumes` page where I think a `Page::ServerVolume` variant would make more sense as it keeps the mapping.

I understand that the page module / update / view is being reused between the volumes route and the server volumes route, and I'm not against having a `Title` trait either so the page models own that data, but I think it makes for sense for there to be separate page variants.